### ISSUE TITLE
Expose Parameter Handler Tag Name

### ISSUE THREAD

**Issue by SimonBoothroyd** [openforcefield/openforcefield#662]
**Is your feature request related to a problem? Please describe.**
It would be useful to publicly expose the `ParameterHandler._TAGNAME` attribute so there is a programatic way to access this for a given class of handler. This would enable e.g.:

```
force_field.get_parameter_handler(ChargeIncrementModelHandler.TAGNAME)
```

**Describe the solution you'd like**
Expose `ParameterHandler._TAGNAME` as `ParameterHandler.TAGNAME` or similar.

**Comment by mattwthompson** [comment]
I've found myself doing this as well (https://github.com/openforcefield/openff-system/search?q=_TAGNAME&unscoped_q=_TAGNAME) and agree  that exposing it would be useful. Maybe as `ParameterHandler.handler_name` or something because the noun "tag" is not used much, if at all, in the public API. Any grounds for hesitation here @j-wags? Otherwise I will implement this

**Comment by j-wags** [comment]
This sounds good, though I might stick with exposing it as `ParameterHandler.TAGNAME`. The "Tag" is from [the XML element nomenclature](https://www.tutorialspoint.com/xml/xml_tags.htm). It's a more XML-centric word than I'd like, but it's the reality of the current situation. 

Also, I don't see significant harm in exposing a `setter` for it as well. So let's go ahead and expose that.

**Comment by mattwthompson** [comment]
I'd push for  `.tag_name` to be consistent with PEP8 naming style but perhaps the value of consistency between the internal and exposed attributes overrides that?

**Comment by j-wags** [comment]
> I'd push for .tag_name to be consistent with PEP8 naming style but perhaps the value of consistency between the internal and exposed attributes overrides that?

Good point, let's do `tag_name`. 

> Also, I don't see significant harm in exposing a setter for it as well. So let's go ahead and expose that.

Ah, I'm a fool. This is a class attribute, not an instance attribute. So let's not expose the `setter`.

**Comment by mattwthompson** [comment]
I totally overlooked that as well (and, to be honest, had to do some digging for why that's a problem!)

**Comment by SimonBoothroyd** [comment]
> I'd push for .tag_name to be consistent with PEP8 naming style 

I think for class constants the PEP8 style is all capitals - so probably `TAG_NAME`

**Comment by j-wags** [comment]
Closed by #665
